contracts/src/TeleporterTokenDestination.sol

```diff
@@ -123,17 +138,16 @@ abstract contract TeleporterTokenDestination is
                 destinationBlockchainID: sourceBlockchainID,
                 destinationAddress: tokenSourceAddress,
                 feeInfo: TeleporterFeeInfo({feeTokenAddress: feeTokenAddress, amount: input.primaryFee}),
-                requiredGasLimit: SEND_TOKENS_REQUIRED_GAS,
-                allowedRelayerAddresses: input.allowedRelayerAddresses,
+                requiredGasLimit: firstHopRequiredGas,
+                allowedRelayerAddresses: new address[](0),
                 message: abi.encode(
                     SendTokensInput({
                         destinationBlockchainID: input.destinationBlockchainID,
                         destinationBridgeAddress: input.destinationBridgeAddress,
                         recipient: input.recipient,
                         primaryFee: input.secondaryFee,
                         secondaryFee: 0,
```

Comment:
Should we be using `input.secondaryFee` for the multihop case?

Reply:
The inner struct here is what is used for the multihop case. The real "secondary fee" is set in the parameters to the external call, and this inner struct is what's used on the destination to send the second message, if necessary, which is why `input.secondaryFee` ends up in the `primaryFee` slot.

Reply:
Ah, got it. Thanks for clarifying.